third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/server/vmService/DartVmServiceDebugProcessZ.java

```diff
@@ -258,12 +276,99 @@ private void onConnectSucceeded(VmService vmService) {
     remoteDebug = true;
 
     vmService.addVmServiceListener(vmServiceListener);
+    myVmOpenSourceLocationListener.addListener(new VmOpenSourceLocationListener.Listener() {
+      @Override
+      public void onRequest(@NotNull String isolateId, @NotNull String scriptId, int tokenPos) {
+        onOpenSourceLocationRequest(isolateId, scriptId, tokenPos);
+      }
+    });
 
     myVmConnected = true;
     getSession().rebuildViews();
     onVmConnected(vmService);
   }
 
+  private void onOpenSourceLocationRequest(@NotNull String isolateId, @NotNull String scriptId, int tokenPos) {
+    myVmServiceWrapper.getObject(isolateId, scriptId, new GetObjectConsumer() {
+      @Override
+      public void received(Obj response) {
+        if (response != null && response instanceof Script) {
+          ApplicationManager.getApplication().executeOnPooledThread(() -> {
+            final XSourcePosition source =
+              getSourcePosition(isolateId, (Script)response, tokenPos);
+            if (source != null) {
+              final Project project = getSession().getProject();
+              final OpenFileHyperlinkInfo
+                info = new OpenFileHyperlinkInfo(project, source.getFile(), source.getLine());
+              ApplicationManager.getApplication().invokeLater(() -> {
+                ApplicationManager.getApplication().runWriteAction(() -> {
+                  info.navigate(project);
+
+                  // TODO(cbernaschina): remove when ProjectUtil.focusProjectWindow(project, true); works as expected.
+                  focusProject(project);
```

Comment:
I think we should make this platform specific. Can you make this conditional on the os (`SystemInfo.isLinux`), and call `ProjectUtil.focusProjectWindow()` in the normal case?